queue/gc_queue.go

```diff
@@ -36,15 +21,15 @@ type gcQueueEntry struct {
 type GCQueue struct {
 	// takeBuffer coordinates the delivery of items taken from the queue
 	// such that they are delivered to requesters.
-	takeBuffer chan Recycler
+	takeBuffer chan interface{}
```

Comment:
Not a blocker, but what's the rationale in downgrading to an `interface{}` (was the `Recycler` struct before) ? I'm guessing in order to be able to consolidate the tests? Even just having be interface be specified in the file makes things a bit more readable IMO.